packages/test/test-service-load/src/runner.ts

```diff
@@ -238,6 +238,7 @@ async function runnerProcess(
 			// If undefined then no fault injection.
 			const faultInjection = runConfig.testConfig.faultInjectionMs;
 			if (faultInjection) {
+				container.once("closed", () => container?.dispose());
```

Comment:
this may work to get around some hangs, but i'm not sure it will work in all cases. if we are going to do a hack like this (which essentially reverses the changes), i'd put it at the top level, right after we create the container, so close always equals dispose. then file a bug to handle the new states correctly in stress, which i'm not sure is possible, and seems like a gap in the design, specifically, the container runtime doesn't know when the container is closed.

Reply:
The problem is that this still leaves hangs possible, as the container may close on its own due to exceptions, it then won't dispose, and the test can hang again.